File: src/Yitter.IdGenTest/GenTest.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using Yitter.IdGenerator;

namespace Yitter.OrgSystem.TestA
{
    public class GenTest
    {
        private IIdGenerator IdGen;
        private Hashtable ids = new Hashtable();
        public IList<long> idList = new List<long>();
        private int GenNumber;
        private int WorkerId;
            

        public GenTest(IIdGenerator idGen, int genNumber, int workerId)
        {
            GenNumber = genNumber;
            IdGen = idGen;
            WorkerId = workerId;
        }

        public void GenId()
        {
            Thread t = new Thread(new ThreadStart(Gen1Start));
            t.Start();
        }

        private void Gen1Start()
        {
            DateTime start = DateTime.Now;

            for (int i = 0; i < GenNumber; i++)
            {
                var id = IdGen.NewLong();
                //ids.Add(id, i);
                idList.Add(id);
            }

            DateTime end = DateTime.Now;
            Console.WriteLine($"++++++++++++++++++++++++++++++++++++++++WorkerId: {WorkerId}, total: {(end - start).TotalSeconds} s");
            Interlocked.Increment(ref Program.Count);
        }
    }
}

File: Java/source/src/main/java/com/github/yitter/test/GenTest.java
package com.github.yitter.test;

import com.github.yitter.contract.IIdGenerator;

import java.util.HashSet;
import java.util.Set;

public class GenTest {

    private IIdGenerator IdGen;
    private int GenIdCount;
    private int WorkerId;
    private Set IdSet = new HashSet();

    public GenTest(IIdGenerator idGen, int genIdCount, int workerId) {
        GenIdCount = genIdCount;
        IdGen = idGen;
        WorkerId = workerId;
    }

    public void GenStart() {
        long start = System.currentTimeMillis();
        long id = 0;

        for (int i = 0; i < GenIdCount; i++) {
            id = IdGen.newLong();
            // IdSet.add(id);
        }

        long end = System.currentTimeMillis();
        long time = end - start;

        System.out.println(id);
        System.out.println("++++++++++++++++++++++++++++++++++++++++WorkerId: "
                + WorkerId + ", total: " + time + " ms");

    }
}
